server/service/src/sync/sync_status.rs

```diff
@@ -0,0 +1,195 @@
+use std::collections::HashMap;
+
+use chrono::{NaiveDateTime, Utc};
+use repository::{
+    DatetimeFilter, RepositoryError, StorageConnection, SyncLogFilter, SyncLogRepository,
+    SyncLogRowRepository,
+};
+use util::Defaults;
+
+#[derive(Debug)]
+pub struct SyncStatus {
+    pub started: Option<NaiveDateTime>,
+    pub finished: Option<NaiveDateTime>,
+}
+
+#[derive(Debug)]
+pub struct SyncStatusWithProgress {
+    pub started: Option<NaiveDateTime>,
+    pub finished: Option<NaiveDateTime>,
+    pub total_progress: u32,
+    pub done_progress: u32,
+}
+
+#[derive(Debug)]
+pub struct FullSyncStatus {
+    pub is_syncing: bool,
+    pub error: Option<String>,
+    pub summary: SyncStatus,
+    pub prepare_initial: Option<SyncStatus>,
+    pub integration: Option<SyncStatus>,
+    pub pull_central: Option<SyncStatusWithProgress>,
+    pub pull_remote: Option<SyncStatusWithProgress>,
+    pub push: Option<SyncStatusWithProgress>,
+}
+
+#[derive(Eq, PartialEq, Hash, Copy, Clone)]
+pub enum SyncStatusType {
+    Initial,
+    Push,
+    PullCentral,
+    PullRemote,
+    Integration,
+}
+
+pub fn is_initialised(connection: &StorageConnection) -> Result<bool, RepositoryError> {
+    match SyncLogRepository::new(&connection).query_by_filter(
```

Comment:
Good idea to outsource as much as possible to db `query_one` would do it. Also we want to query by `done_endtime` 

`prepare_initial_done_datetime` only indicates that we've asked the server to prepare initial remote records